pkg/authorization/local_authorizer.go

```diff
@@ -33,26 +33,38 @@ import (
 )
 
 type LocalAuthorizer struct {
-	roleLister               rbacv1listers.RoleClusterLister
-	roleBindingLister        rbacv1listers.RoleBindingClusterLister
-	clusterRoleBindingLister rbacv1listers.ClusterRoleBindingClusterLister
-	clusterRoleLister        rbacv1listers.ClusterRoleClusterLister
```

Comment:
these were intentional I believe to ensure the informers are started. If you move them into the hot path, the `Lister()` call might come to later after the shared informer factory has been started. Creates ugly races.

Reply:
code comment would be good if this is the case

Reply:
Yeah, let me bring this back and add a note. I wasn't aware of this and thought they were just written at different times and there was room to unify the code.

Reply:
I'll do a little PR history search if I can find some additional context.

Reply:
we do similar things in the controllers. They all store listers in their struct to enforce creation of the informers at constructor time.

Reply:
Question here: Should we adjust the `GlobalAuthorizer` as well then? Because I copied this pattern from it, so it would be susceptible to the same races, no? Is there something with the `localKubeInformers` and `globalKubeInformers` passed to _that_ that makes it okay?

Reply:
Deleted my previous comment by mistake. I copied this pattern over from the `GlobalAuthorizer`, which also takes `SharedInformerFactory` input parameters. Should we adjust that one as well, or is there something special with it that will prevent those races mentioned?